util/cli.cr

```diff
@@ -0,0 +1,304 @@
+require "./readme"
+require "yaml"
+require "json"
+require "http/client"
+require "colorize"
+require "option_parser"
+
+GITLAB_CLIENT     = HTTP::Client.new "gitlab.com", tls: true
+TRAVIS_CLIENT     = HTTP::Client.new "api.travis-ci.org", tls: true
+TRAVIS_PRO_CLIENT = HTTP::Client.new "api.travis-ci.com", tls: true
+CIRCLE_CLIENT     = HTTP::Client.new "circleci.com", tls: true
+DRONE_CLIENT      = HTTP::Client.new "cloud.drone.io", tls: true
+
+# Represents a version control provider.
+enum VC
+  Github
+  Gitlab
+
+  # Returns the protocal and domain for `self`.
+  def host : String
+    case self
+    when .github? then "https://github.com"
+    when .gitlab? then "https://gitlab.com"
+    end.not_nil!
+  end
+end
+
+# Represents a CI provider.
+#
+# TODO: Add a way to resolve `Actions`.
+enum CI
+  None
+  Travis
+  Gitlab
+  Circle
+  Actions
+  Drone
+  TravisPro
+  Exempt
+end
+
+# Represents a shard within `README.md`.
+class Shard
+  include YAML::Serializable
+
+  # Manually keep track of shards using `CI::Actions` since there
+  # is not currently a way to resolve if a shard uses it.
+  private USING_GH_ACTIONS = {
+    "blacksmoke16/assert",
+    "soveran/toro",
+    "soveran/ohm-crystal",
+    "soveran/stal-crystal",
+    "soveran/resp-crystal",
+  }
+
+  # Shards that are exempt from reporting.
+  # Mainly for utilities and plugins that
+  # can't really be tested autonomously.
+  private EXEMPT_SHARDS = {
+    "dotmilk/emacs-crystal-mode",
+    "veelenga/crystal-zsh",
+  }
+
+  # The owner of the shard's repo.
+  #
+  # Can either be a user or organization.
+  getter owner : String
+
+  # The name of the shard's repo.
+  getter name : String
+
+  # Which `VC` `self` uses.
+  getter vc : VC
```

Comment:
```yml
- owner: veelenga
  name: crystal-zsh
  vc: 0   # -> here
  ci: 7    # -> and here
  last_build: 2019-11-02 17:22:17.981983000
```

I would probably use some converter to see strings in the generated file rather than integers. 

Not a blocker.